### ISSUE TITLE
Can I apply Custom Dataset?

### ISSUE THREAD

**Issue by LJKwang** [jhgan00/image-retrieval-transformers#7]
Hello, I'm trying to learn by applying a custom dataset with unsupervised learning without labeling, is it possible? 
If It's possible, could you tell me how?

**Comment by jhgan00** [comment]
The IRT basically requires a labeled dataset. It would be more appropriate to refer to [MAE](https://github.com/facebookresearch/mae) or [SimCLR](https://github.com/sthalles/SimCLR). These methods may require a relatively large scale of data and batch sizes.

**Comment by LJKwang** [comment]
I have an additional question!

1. I'd like to see multiple renking lists like the test code here, can I apply the same for MAE or simCLR?

2. SOP DATASET doesn't have a label but only has a class id, so isn't this a non-supervised learning?

Thank you for reading and responding!

**Comment by jhgan00** [comment]
1. Yes, you can extract compressed representations of images using trained models and retrieve similar images in the embedding space based on metrics like cosine similarity.


2. Could you please clarify the term 'label'? The class IDs of the SoP dataset are a type of labels. https://paperswithcode.com/dataset/stanford-online-products.

**Comment by jhgan00** [comment]
The settings defined in args are primarily parameters that only affect the training phase. If you are only running inference using a trained model, there should be no issues.

However, if you intend to continue training the model, you might need to adjust the hyperparameters that are defined in args to achieve better performance.

**Comment by LJKwang** [comment]
Thank you very much!

**Comment by jhgan00** [comment]
Please feel free to leave a comment if you need more support
